src/Microsoft.Health.Fhir.Core/Features/Definition/CompartmentIndexer.cs

```diff
@@ -74,7 +74,7 @@ public CompartmentIndices Extract(string resourceType, IReadOnlyCollection<Searc
                 {
                     string key = compartmentType.ToString();
 
-                    if (!retDict.TryGetValue(key, out List<SearchIndexEntry> searchIndexEntries))
+                    if (!retDict.ContainsKey(key))
```

Comment:
should this remain ` if (!retDict.TryGetValue(key, out List<SearchIndexEntry> searchIndexEntries))` but change line to be

```
searchIndexEntries = new List<SearchIndexEntry>();
retDic[key] = searchIndexEntries;
```

then change line 82 to be `searchIndexEntries.Add(indexEntry)`?